### PR TITLE
TaegisXDR - Add support for comments

### PR DESCRIPTION
**Pull request by reedium**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
* Adds support for creating, updating, and fetching comments
* Replaces `readable_output` with tableToMarkdown
* Added proper errors when unable to fetch investigation or playbook execution
* Updated documentation for context data
* Updated valid Investigation statuses

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [x] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by ShacharKidor** [comment]
Hi @reedium,
Thank you for your contribution!
Your PR review will be slightly delayed as we are at the start of the holiday season -
I apologize in advance.

Thanks again :)

File: Packs/SecureWorks/ReleaseNotes/4_1_0.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+
+#### Integrations
+##### TaegisXDR
+- Added the following new commands:
+    - ***taegisxdr-create-comment***
+    - ***taegisxdr-fetch-comment***
+    - ***taegisxdr-fetch-comments***
+    - ***taegisxdr-update-comment***
+- Added ***url*** field output for reference in most commands.
```
Comment (moishce) [inline Packs/SecureWorks/ReleaseNotes/4_1_0.md:9]: Where is this done in the code?

File: Packs/SecureWorks/ReleaseNotes/4_1_0.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+- Added test for test-module command.
+- Added errors when an investigation or playbook execution cannot be found.
+- Replaced the readable output of all commands with Markdown tables.
+- Updated documentation related to Context Data of all commands.
+    - Context path was updated in 4.0.0 but the documentation had not been updated to reflect the changes.
```
Comment (moishce) [inline Packs/SecureWorks/ReleaseNotes/4_1_0.md:14]:
```suggestion
```
I don't think it needs to be written in the release notes

**Review by moishce — COMMENTED** [review]
I appreciate your contribution, thank you
You have done a great job!

File: Packs/SecureWorks/ReleaseNotes/4_1_0.md (hunk 1 above)
Comment (reedium) [reply Packs/SecureWorks/ReleaseNotes/4_1_0.md:9]: Sure thing. If you're referring to the `url` field, an example of this is can be seen, on line Line 875 of TaegixXDR.py. This is to work with the `url_keys` of the `tableToMarkdown` method. 

I did this everywhere it made sense. If needed, I can remove this from the release notes.

File: Packs/SecureWorks/ReleaseNotes/4_1_0.md (hunk 2 above)
Comment (reedium) [reply Packs/SecureWorks/ReleaseNotes/4_1_0.md:14]: Sure, I can remove that once we confirm we're okay with the previous comment.

File: Packs/SecureWorks/ReleaseNotes/4_1_0.md (hunk 1 above)
Comment (reedium) [reply Packs/SecureWorks/ReleaseNotes/4_1_0.md:9]: @moishce Does this makes sense? I'd like to get this PR completed as soon as possible as we have a few users awaiting these changes.

**Comment by reedium** [comment]
@moishce I attempted to sync the branch in my github with master (as CI was failing) but it looks like it pulled in Michal as a contributor to the PR for some reason.

Let me know the best way to solve this. I can recreate my branch/commits and resubmit the PR if needed, although I'd rather not if possible.

File: Packs/SecureWorks/ReleaseNotes/4_1_0.md (hunk 1 above)
Comment (moishce) [reply Packs/SecureWorks/ReleaseNotes/4_1_0.md:9]: I think it can be removed from the release notes
It can be confusing

Comment (reedium) [reply Packs/SecureWorks/ReleaseNotes/4_1_0.md]: Sure thing. I've removed it from the release notes.

File: Packs/SecureWorks/ReleaseNotes/4_1_0.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+
+#### Integrations
+##### TaegisXDR
+- Added the following new commands:
+    - ***taegisxdr-create-comment***
+    - ***taegisxdr-fetch-comment***
+    - ***taegisxdr-fetch-comments***
+    - ***taegisxdr-update-comment***
+- Added test for test-module command.
```
Comment (moishce) [inline Packs/SecureWorks/ReleaseNotes/4_1_0.md:9]:
```suggestion
```
This can be deleted because it does not affect the user.

**Review by moishce — APPROVED** [review]

**Pull request merged**